From cdd74f2b1abed71f12de65cfc1495153635d02e5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 23 Dec 2021 07:33:04 +0200
Subject: [PATCH] [SW-0] gaudi2: scrub arc1 iccm memory

In order to avoid any ECC exception observed on ARC memories, it is
recommended to scrub both ICCM and DCCM memories. Since ARC0 memories
scrubbing is already done and ARC1 DCCM is scrubbed as part of ROM,
this patch add changes to scrub ARC1 ICCM memory conditionally on
cold reset.

Change-Id: I90e78d45a18f6066c7ca7c75a1c121d2bf2046e9
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 1f6e765..13ae454 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -131,8 +131,16 @@
 
 void hl_plat_init(void)
 {
+	u32 rst_src;
+
 	/* need to configure arb_timeout val before PLL is locked */
 	set_slv_cfg_reg_val();
+
+	rst_src = hl_rst_src_get();
+	if (rst_src != HL_COLD_RST)
+		return;
+
+	init_arc_mem(1, ARC1_ICCM_FULL_BASE, ARC1_ICCM_SIZE);
 }
 
 void preboot_copy_image(u64 dst, const void *buf, int len, const char *arch)
